Extension/test/unit/commandLineParsing.test.ts

```diff
@@ -0,0 +1,217 @@
+
+import { deepEqual, throws } from 'assert';
+import { describe } from 'mocha';
+import { extractArgs } from '../../src/Utility/Process/commandLine';
+import { is } from '../../src/Utility/System/guards';
+import { isWindows } from '../../src/constants';
+import { when } from '../common/internal';
+
+// eslint-disable-next-line import/no-unassigned-import
+require('source-map-support/register');
+
+function marker() {
+    try {
+        throw new Error('Test Marker');
+    } catch (E) {
+        if (is.error(E)){
+            return E.stack?.split('\n').filter(each => each.includes('.ts') && each.includes('<anonymous>')).join('\n');
+        }
+    }
+}
+
+const fails: [ string | undefined, string, string?][] = [
+    /** command substitution not supported */
+    [undefined, "explicit ``", marker()],
+    [undefined, "$(echo hello)", marker()],
+    [undefined, "$( (echo hello) )", marker()],
+    [undefined, "$((echo hello);(echo there))", marker()],
+    [undefined, "`echo one two`", marker()],
+    [undefined, "$(echo ')')", marker()],
+    [undefined, "$(echo hello; echo)", marker()],
+    [undefined, "a$(echo b)c", marker()],
+    ["pepperoni", "${var%$(echo oni)}",  marker()],
+    [undefined, "\"$(echo hello there)\"",  marker()],
+    [undefined, "\"$(echo \"hello there\")\"",  marker()],
+    ["1", "$(( $(echo 3)+$var ))", marker()],
+    [undefined, "\"$(echo \"*\")\"",  marker()],
+    [ undefined, "\"a\n\n$(echo)b\"",  marker()],
+    /* Things that should fail */
+    [ undefined, "new\nline", marker()],
+    [ undefined, "pipe|symbol", marker()],
+    [ undefined, "&ampersand", marker()],
+    [ undefined, "semi;colon", marker()],
+    [ undefined, "<greater", marker()],
+    [ undefined, "less>", marker()],
+    [ undefined, "(open-paren", marker()],
+    [ undefined, "close-paren)", marker()],
+    [ undefined, "{open-brace", marker()],
+    [ undefined, "close-brace}", marker()],
+    [ undefined, "$(ls)", marker()],
+    [ undefined, "${50+20))", marker()],
+    [ undefined, "${%%noparam]", marker()],
+    [ undefined, "${missing-brace", marker()],
+    [ undefined, "$(for i in)", marker()],
+    [ undefined, "$((2+))", marker()],
+    [ undefined, "`", marker()],
+    [ undefined, "$((010+4+))", marker()],
+    /* Test for CVE-2014-7817. We test 3 combinations of command
+       substitution inside an arithmetic expression to make sure that
+       no commands are executed and error is returned.  */
+    [ undefined, "$((`echo 1`))", marker() ],
+    [ undefined, "$((1+`echo 1`))", marker() ],
+    [ undefined, "$((1+$((`echo 1`))))", marker() ],
+
+    [ undefined, "`\\", marker() ],     /* BZ 18042  */
+    [ undefined, "${", marker() ],      /* BZ 18043  */
+    [ undefined, "L${a:", marker()],   /* BZ 18043#c4  */
+    [ undefined, "${1/0]", marker() ] /* BZ 18100 */
+];
+
+const success: [string | undefined, string, string[], string?][] = [
+    /* Simple word- and field-splitting */
```

Comment:
From what I can tell, the `-` should be removed.

Reply:
The comment came from the original test suite for wordexp, but I can fix it.